libr/anal/fcn.c

```diff
@@ -2463,7 +2463,22 @@ R_API bool r_anal_fcn_get_purity(RAnal *anal, RAnalFunction *fcn) {
 }
 
 static bool can_affect_bp(RAnal *anal, RAnalOp* op) {
-	return op->dst && op->dst->reg && op->dst->reg->name &&!strcmp (op->dst->reg->name, anal->reg->name[R_REG_NAME_BP]); 
+	RAnalValue *dst = op->dst;
+	RAnalValue *src = op->src[0];
+	char *opdreg = NULL;
+	char *opsreg = NULL;
+	const char *bp_name = anal->reg->name[R_REG_NAME_BP];
+	if (dst && dst->reg) {
```

Comment:
assign and define in the same line with ternary operator, it will b shorter and easieer to read